A6lib.cpp

```diff
@@ -368,99 +417,113 @@ void A6lib::enableSpeaker(byte enable) {
 // Autodetect the connection rate.
 
 long A6lib::detectRate() {
-    unsigned long rate = 0;
-    unsigned long rates[] = {9600, 115200};
-
-    // Try to autodetect the rate.
-    logln("Autodetecting connection rate...");
-    for (int i = 0; i < countof(rates); i++) {
-        rate = rates[i];
-
-        A6conn->begin(rate);
-        log("Trying rate ");
-        log(rate);
-        logln("...");
-
-        delay(100);
-        if (A6command("\rAT", "OK", "+CME", 2000, 2, NULL) == A6_OK) {
-            return rate;
-        }
-    }
-
-    logln("Couldn't detect the rate.");
-
-    return A6_NOTOK;
+	unsigned long rate = 0;
+	unsigned long rates[] = { 9600, 115200 };
+
+	// Try to autodetect the rate.
+	LOG("Autodetecting connection rate...");
+	for (int i = 0; i < countof(rates); i++) {
+		rate = rates[i];
+
+		Serial1.printf("sumping ports");
+		if (ports.hport != nullptr)
+			Serial.println("hardware ports is present....");
+		if (ports.sport != nullptr)
+			Serial1.println("software port is present...");
+
+		if (ports.hport != nullptr)
+			ports.hport->begin(rate);
```

Comment:
Couldn't this go in the `if` checks above?

Reply:
I Fix all these in next commits and it's perfect:)
I recommend to delay this review until next one.